plugins/modules/setup.ps1

```diff
@@ -559,17 +559,29 @@ $factMeta = @(
     @{
         Subsets = 'all_ipv4_addresses', 'all_ipv6_addresses'
         Code = {
-            $interfaces = [System.Net.NetworkInformation.NetworkInterface]::GetAllNetworkInterfaces()
-            $ips = @(foreach ($interface in $interfaces) {
-                    # Win32_NetworkAdapterConfiguration did not return the local IPs so we replicate that here.
-                    $interface.GetIPProperties().UnicastAddresses | Where-Object {
-                        $_.Address.ToString() -notin @('::1', '127.0.0.1')
-                    } | ForEach-Object -Process {
-                        $_.Address.ToString()
+            $NetIPAddress = Get-CimInstance -Namespace ROOT/StandardCimv2 -ClassName MSFT_NetIPAddress
+
+            $ips = @()
```

Comment:
This is a minor nit but it might be nicer to use a generic list here instead of `+=` with an array. At the number of ips being added it's mostly inconsequential but I find it always nice to avoid `+=` on arrays regardless.